ml.q

```diff
@@ -258,12 +258,18 @@ cgroup:{[df;X;C]value group imin f2nd[df X] C}
 / matri(X) to the nearest (C)entroid and then uses the (c)entroid (f)unction
 / to update the centroid location.
 lloyd:{[df;cf;X;C]cf X@\: cgroup[df;X;C]}
+/ uses (r)esponsibility (f)unction David Mackay's Information Theory... pg 289
+lloyds:{[df;cf;rf;X;C] cf[rf .ml.f2nd[df X] C;X]} /soft assignment 
 
 kmeans:lloyd[edist2;avg'']      / k-means
 kmedians:lloyd[mdist;med'']     / k-medians
 khmeans:lloyd[edist2;hmean'']   / k harmonic means
 skmeans:lloyd[cosdist;normalize (avg'')::] / spherical k-means
 
+kmeanss:lloyds[edist2;wavg\:/:;{x=\:min x}] /k-means using loyd with rf
```

Comment:
looks like `{x=\:min x}` can be replaced with `dax[=;min]` and the code will then use `peach`

Reply:
agree, I didn't think of that function, fixed; though dax is not as transparent to me as x=\:min